data_model/src/transaction.rs

```diff
@@ -229,6 +247,143 @@ model! {
+        Serialize,
+        IntoSchema,
+    )]
+    #[display(fmt = "{self:?}")] // TODO ?
+    #[ffi_type]
+    pub struct SignedTransaction {
+        /// [`Transaction`] payload.
+        pub payload: TransactionPayload,
+        /// [`SignatureOf`]<[`TransactionPayload`]>.
+        pub signatures: btree_set::BTreeSet<SignatureOf<TransactionPayload>>,
+    }
+
+    /// Transaction Value used in Instructions and Queries
+    #[derive(
+        Debug, Clone, PartialEq, Eq, Hash, Decode, Encode, Deserialize, Serialize, IntoSchema,
+    )]
+    #[ffi_type(local)]
+    pub enum TransactionValue {
+        /// Committed transaction
+        Transaction(Box<VersionedSignedTransaction>),
+        /// Rejected transaction with reason of rejection
+        RejectedTransaction(Box<VersionedRejectedTransaction>),
+    }
+
+    /// `TransactionQueryResult` is used in `FindAllTransactions` query
+    #[derive(
+        Debug,
+        Clone,
+        PartialEq,
+        Eq,
+        Hash,
+        Getters,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+    )]
+    #[getset(get = "pub")]
+    #[ffi_type]
+    pub struct TransactionQueryResult {
+        /// Transaction
+        pub tx_value: TransactionValue,
+        /// The hash of the block to which `tx` belongs to
+        pub block_hash: Hash,
+    }
+
+    /// `ValidTransaction` represents trustfull Transaction state.
+    #[version_with_scale(n = 1, versioned = "VersionedValidTransaction")]
+    #[derive(
+        Debug, Clone, PartialEq, Eq, Hash, Decode, Encode, Deserialize, Serialize, IntoSchema,
+    )]
+    #[ffi_type]
+    pub struct ValidTransaction {
+        /// The [`Transaction`]'s payload.
+        pub payload: TransactionPayload,
+        /// [`SignatureOf`]<[`TransactionPayload`]>.
+        pub signatures: SignaturesOf<TransactionPayload>,
+    }
+
+    /// [`RejectedTransaction`] represents transaction rejected by some validator at some stage of the pipeline.
+    #[version(n = 1, versioned = "VersionedRejectedTransaction")]
+    #[derive(
+        Debug,
+        Clone,
+        PartialEq,
+        Eq,
+        Hash,
+        Getters,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+    )]
+    #[ffi_type]
+    pub struct RejectedTransaction {
+        /// The [`Transaction`]'s payload.
+        pub payload: TransactionPayload,
+        /// [`SignatureOf`] [`Transaction`].
+        pub signatures: SignaturesOf<TransactionPayload>,
+        /// The reason for rejecting this transaction during the validation pipeline.
+        #[getset(get = "pub")]
+        pub rejection_reason: error::TransactionRejectionReason,
+    }
+
+    /// `AcceptedTransaction` — a transaction accepted by iroha peer.
+    #[version_with_scale(n = 1, versioned = "VersionedAcceptedTransaction")]
+    #[derive(Debug, Clone, Decode, Encode, Serialize)]
+    #[ffi_type]
+    pub struct AcceptedTransaction {
```

Comment:
previously it was `pub(crate)`. This was a feature where a struct would be turned private if built without `transparent_api`. In this case this transaction is useless to `iroha_client` or `iroha_wasm` so it was removed from public API.

Reply:
the same was with `QueryPayload` and `AcceptTransactionFailure`

Reply:
So they shouldn't be `ffi_type` then?

Reply:
no